compiler/resolution/virtualDispatch.cpp

```diff
@@ -390,7 +390,25 @@ static void addToVirtualMaps(FnSymbol* pfn, AggregateType* ct) {
               USR_FATAL_CONT(pfn, "conflicting return type specified for '%s: %s'", toString(pfn), pfn->retType->symbol->name);
               USR_FATAL_CONT(fn, "  overridden by '%s: %s'", toString(fn), fn->retType->symbol->name);
               USR_STOP();
+            } else if (fn->throwsError() != pfn->throwsError()) {
```

Comment:
If the parent class function throws, then I think the child class function could be throws or not throws. In a way, a non-throwing function is a subtype of a throwing function. What do you think?

Reply:
I agree that a non-throwing function can be viewed as a subtype of a throwing one. However, I'm uncomfortable with the implementation changes we'd have to do at this time.

In particular, in the implementation of virtual dispatch, what we have is a table from runtime class ID to the method that should be called. If we just allowed it, we would pass an extra argument (out error) to a dynamically called function. That seems to be at least surprising and potentially an error. (Do you know if when calling a function through a function pointer, it is always OK to pass extra arguments not expected by the function?)

Reply:
This might be a question for @dmk42? Theoretically the arguments get pushed on the stack in reverse order so the later arguments just get ignored, but that might just be happy coincidence rather than something spec enforced.

Reply:
When targeting C, if the number of arguments does not equal the expected number, it is undefined behavior.  In practice, most ABIs end up just ignoring an extra argument as long as it's tacked onto the end, though we can't count on that.  When targeting LLVM, we have more control and could set things up to ensure extra arguments just get ignored, if we want to do it that way.

Reply:
@dmk42 - thanks for the clarification. I think it's sufficient for it to be an error for now & we can consider relaxing it in the future.